### DISCUSSION TITLE
MDX Net - Segment size and Overlap? What do they mean?

### DISCUSSION THREAD

**Discussion by deadniell** [Anjok07/ultimatevocalremovergui#831] [Q&A]
How do you even configure that?

**Comment by dogtopus** [comment]
Segment size controls how large each chunk of audio data gets loaded onto the GPU. The bigger it is, the less time you spend on swapping data between RAM and VRAM, thus the faster the processing. Also the bigger the segment size, the less it needs to rely on overlapping, thus might give better results. Set it to as large as your VRAM allows.

Overlap presumably controls how much the audio data chunks are overlapped. I just leave it as default and probably won't change it unless there are artifacts in the result that I suspect to be caused by insufficient context given to the model (like gaps in the audio).

**Reply by andrixpan** [reply]
i use gpu rtx 2060 6gb vram and ram pc 16gb, 
what segment size should it be?

**Reply by dogtopus** [reply]
Just test it yourself. Start from a size, run with a long audio clip (\~20mins), check nvidia-smi, interpolate the size based on VRAM usage, rinse and repeat until the VRAM is almost filled (I'd say 90-95% would be a good goal to hit).

**Reply by dgoryeo** [reply]
Does 256 segment size mean 256 bytes? Or mega bytes?

**Reply by jordigoyanes** [reply]
@dogtopus I have the same question as @dgoryeo

**Reply by dogtopus** [reply]
> Does 256 segment size mean 256 bytes? Or mega bytes?

It's probably some internal unit, so neither.

**Reply by ghost** [reply]
segment size value is the number of STFT frames by audio chunk. 

Here are some overlap/segment size benchmarks made by Bas Curtiz from Audio Separation Discord community:
![image](https://github.com/Anjok07/ultimatevocalremovergui/assets/6044199/51c86960-aee8-44f8-b229-bf018ba9609b)

**Reply by dgoryeo** [reply]
Then as a reference/guideline this can be a method to estimate VRAM useage roughly: 

For a 1-second mono audio signal sampled at 44.1 kHz, with a frame size of 1024 samples and 50% overlap, and using a 1024-point FFT:

The audio has 44,100 samples.
With 50% overlap, each frame overlaps the previous one by 512 samples, so the hop size (step between consecutive frames) is 512 samples.
The number of frames is approximately (44100 - 1024) / 512 + 1 ≈ 86 frames.
Memory usage is roughly 86 frames * 1024 FFT points * 16 bytes ≈ 1.39 MB per channel.

**Reply by scryptkyddy** [reply]
TLDR: for example, MDX23C - instvocD1581  GPU mode, with 6GB 1060 GPU (P106-100) and R3 Quad core 3100, setting **segments to 736** and **overlap to 50**  (max) uses just about **5.5GB of the 6GB** on the GPU, takes about 15 minutes  for a 4 minute 1970s song, but results are great.
![uvr lovely dayCapture](https://github.com/user-attachments/assets/d75dfa50-5b2d-4edd-a6f6-5f0f2b2f7fe5)

Also, say you wanted to speed it up at the cost of track bleed/artifacts between tracks, keep segments at about 736 (for 6gb GPU) and reduce overlap to something less than 50 (extreme quality). In my case i'm splitting just the vocals out now for the next step of 6 stem separation (later).

**Reply by Buvi07** [reply]
use UVR-MDX-NET Inst Main model its very fast and accuracy level is greater for its speed
